src/NServiceBus.Persistence.AzureTable/SagaPersisters/SagaIdGenerator.cs

```diff
@@ -32,16 +36,73 @@ public static Guid Generate(string sagaEntityTypeFullName, SagaCorrelationProper
             return DeterministicGuid($"{sagaEntityTypeFullName}_{correlationProperty.Name}_{serializedPropertyValue}");
```

Comment:
@danielmarbach I assume that you are optimizing for allocations? Could this string concatenation be prevented? Then `.HashCore` and `.HashFinal` could be used with the input string parts.

Reply:
Yes it could be. See https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/389
I was planning to do that as a dedicated additional PR

Reply:
@danielmarbach exactly, but I see that uses a string buffer to copy data which is not needed when using `.HashCore` / `.HashFinal`. Not sure on the performance though but I think it simplifies the code.

Reply:
Wasn't aware of HasCore and HasFinal. Need to do some reading there first. In case you have some good sources please share.

Reply:
@danielmarbach Unfortunately the MS docs are non existing on hashing implementations, usage and performance. You can look at the implementation of `ComputeHash` which uses these APIs too if I'm not mistaken.

Reply:
@ramonsmits I couldn't find those properties in the runtime repo with searching. Can you maybe point me in the right direction?

Reply:
@danielmarbach 

- https://docs.microsoft.com/lt-lt/dotnet/api/system.security.cryptography.hashalgorithm.hashfinal
- https://docs.microsoft.com/lt-lt/dotnet/api/system.security.cryptography.hashalgorithm.hashcore